File: storm-core/src/test/java/st/orm/core/model/polymorphic/IntDiscAnimal.java
package st.orm.core.model.polymorphic;

import st.orm.DbTable;
import st.orm.Discriminator;
import st.orm.Discriminator.DiscriminatorType;
import st.orm.Entity;

/**
 * Single-Table Inheritance with INTEGER discriminator type.
 */
@Discriminator(type = DiscriminatorType.INTEGER)
@DbTable("int_disc_animal")
public sealed interface IntDiscAnimal extends Entity<Integer> permits IntDiscCat, IntDiscDog {
}

File: storm-kotlin/src/test/kotlin/st/orm/template/model/IntDiscAnimal.kt
package st.orm.template.model

import st.orm.DbTable
import st.orm.Discriminator
import st.orm.Discriminator.DiscriminatorType.INTEGER
import st.orm.Entity
import st.orm.PK

@Discriminator(type = INTEGER)
@DbTable("int_disc_animal")
sealed interface IntDiscAnimal : Entity<Int>

@Discriminator("1")
data class IntDiscCat(
    @PK val id: Int = 0,
    val name: String,
    val indoor: Boolean,
) : IntDiscAnimal

@Discriminator("2")
data class IntDiscDog(
    @PK val id: Int = 0,
    val name: String,
    val weight: Int,
) : IntDiscAnimal
